builtin/providers/cloudstack/tags.go

```diff
@@ -0,0 +1,81 @@
+package cloudstack
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+// tagsSchema returns the schema to use for tags.
+//
+func tagsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeMap,
+		Optional: true,
+	}
+}
+
+// setTags is a helper to set the tags for a resource. It expects the
+// tags field to be named "tags"
+func setTags(cs *cloudstack.CloudStackClient, d *schema.ResourceData, resourcetype string) error {
+
+	if d.HasChange("tags") {
+		oraw, nraw := d.GetChange("tags")
+		o := oraw.(map[string]interface{})
+		n := nraw.(map[string]interface{})
+		create, remove := diffTags(tagsFromMap(o), tagsFromMap(n))
```

Comment:
But it will never be more then the two API calls you already make right? You make 1 call with all the tags that need to be removed and 1 call with all the tags that need to be created.

That's exactly the same amount of calls you have now right? Or am I missing something?